core/src/xml.rs

```diff
@@ -1,6 +1,41 @@
-//! Garbage-collectable XML DOM impl
+use regress::Regex;
+use std::sync::OnceLock;
 
-mod iterators;
-mod tree;
+static ENTITY_REGEX: OnceLock<Regex> = OnceLock::new();
 
-pub use tree::{custom_unescape, XmlNode, ELEMENT_NODE, TEXT_NODE};
+/// Handles flash-specific XML unescaping behavior.
+/// We accept all XML entities, and also accept standalone '&' without
+/// a corresponding ';'
+pub fn custom_unescape(
+    data: &[u8],
+    decoder: quick_xml::Decoder,
+) -> Result<String, quick_xml::Error> {
+    let input = decoder.decode(data)?;
+
+    let re = ENTITY_REGEX.get_or_init(|| Regex::new(r"&[^;]*;").unwrap());
+    let mut result = String::new();
+    let mut last_end = 0;
+
+    // Find all entities, and try to unescape them.
+    // Our regular expression will skip over '&' without a matching ';',
+    // which will preserve them as-is in the output
+    for cap in re.find_iter(&input) {
+        let start = cap.start();
+        let end = cap.end();
+        result.push_str(&input[last_end..start]);
+
+        let entity = &input[start..end];
+        // Unfortunately, we need to call this on each entity individually,
+        // since it bails out if *any* entities in the string lack a terminating ';'
+        match quick_xml::escape::unescape(entity) {
+            Ok(decoded) => result.push_str(&decoded),
+            // FIXME - check the actual error once https://github.com/tafia/quick-xml/pull/584 is merged
```

Comment:
Unrelated to your changes, but this is quite an old unfixed FIXME, https://github.com/tafia/quick-xml/pull/584 was merged over 2 years ago.